File: hw/cañarte/u3/HW28SingletonAndAbstractFactory/Java/TaxSingleton/src/main/java/ec/edu/espe/taxsingleton/USTax.java
package ec.edu.espe.taxsingleton;

/**
 *
 * @author Saray Canarte, Code Creators, DCCO-ESPE
 */
public class USTax {
   
    private static USTax instance;
    
    private USTax() {
        //this is the private constructor
    }
    
    public static USTax getInstance() {
        if (instance == null) {
            instance = new USTax(); 
        }
        return instance;
    }

    public float calculateSalesTax(float saleAmount) {
        float taxRate = 9.38f/100;
        return saleAmount * taxRate;
    }
    public float calculateTaxplusSale(float saleAmount, float tax){
        float saleTotal = saleAmount + tax;
        return saleTotal;
     }
}

File: hw/Rodriguez/u3/hw28/hw28Python/Singleton/taxsingleton/UStax.py
class USTax:
    _instance = None

    def __new__(cls):
        if cls._instance is None:
            cls._instance = super(USTax, cls).__new__(cls)
        return cls._instance

    def calculate_sales_tax(self, sale_amount):
        tax_rate = 9.38 / 100
        return sale_amount * tax_rate

    def calculate_tax_plus_sale(self, sale_amount, tax):
        sale_total = sale_amount + tax
        return sale_total
